 Contexto y Preparación: Son el equipo de Inteligencia Artificial de una firma de análisis móvil. Necesitan entrenar un modelo para predecir el éxito de una app, pero el archivo CSV que les mandó el equipo de Scraping es un desastre de formato. Tienen que convertirlo en una matriz matemática estricta.

Instrucciones Técnicas: Creen un Notebook  nuevo y carga el dataset crudo:

In [70]:
import pandas as pd
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/googleplaystore.csv"
df = pd.read_csv(url)

3. Preguntas de Análisis (Para responder con código y celdas Markdown):

Pregunta A: El scraper cometió errores y guardó la misma app varias veces. Escriban una línea de código para identificar si hay nombres duplicados en la columna App (df.duplicated(subset=['App']).sum()). Si los hay, apliquen .drop_duplicates(subset=['App'], keep='first', inplace=True) para quedarse solo con la primera aparición de cada aplicación. ¿Cuántos registros duplicados purgaron?

In [71]:
dup_data = df.duplicated(subset=['App']).sum()
print(f"Registros duplicados encontrados: {dup_data}")

dup_data = df.drop_duplicates(subset=['App'], keep='first', inplace=True)
print(f"Registros tras eliminarlos: {dup_data}")

Registros duplicados encontrados: 1181
Registros tras eliminarlos: None


Se purgaron 1181 registros duplicados

Pregunta B: La columna Installs tiene texto como "1,000,000+". Un algoritmo no puede procesar eso.
Escriban una instrucción encadenando el método .str.replace() para eliminar el símbolo + y otro para eliminar la coma ,. (Ojo: utilicen regex=False o escapen el símbolo más \+).
Al final de la cadena, utilicen .astype(int) (o float si tienen NaNs) para convertir la columna.
Comprueben el éxito ejecutando df['Installs'].mean(). ¿Cuál es el promedio real de descargas?

In [72]:
df = df[df['Installs'] != 'Free']

df['Installs'] = (
    df['Installs']
    .str.replace('+', '', regex=False)
    .str.replace(',', '', regex=False)
    .astype(int)
)

down_avg = df['Installs'].mean()
print(f"Promedio de descargas: {down_avg}")


Promedio de descargas: 7777506.732270421


Pregunta C: La columna Price tiene valores como "$$4.99" y "0". Apliquen la misma lógica arquitectónica que en la Pregunta B: eliminen el símbolo de dólar $ y conviertan la columna usando .astype(float).

In [73]:
df['Price'] = (
    df['Price']
    .str.replace('$', '', regex=False)
    .astype(float)
)

df['Price'].describe()

# price_avg = df['Price'].mean()
# print(f"Promedio de descargas: {price_avg}")


count    9659.000000
mean        1.099299
std        16.852152
min         0.000000
25%         0.000000
50%         0.000000
75%         0.000000
max       400.000000
Name: Price, dtype: float64

Pregunta D: Ahora que el precio es numérico, ejecuten df['Price'].max(). Notarán que la app más cara cuesta alrededor de 400 dólares.

In [74]:
df['Price'].max()

np.float64(400.0)

a. Usen un filtro booleano (df[df['Price'] > 200]) para descubrir cuáles son estas apps. Verán que son aplicaciones basura o de broma.

In [75]:
df[df['Price'] > 200].value_counts()

App                             Category   Rating  Reviews  Size  Installs  Type  Price   Content Rating  Genres         Last Updated       Current Ver  Android Ver 
I AM RICH PRO PLUS              FINANCE    4.0     36       41M   1000      Paid  399.99  Everyone        Finance        June 25, 2018      1.0.2        4.1 and up      1
I Am Rich                       FAMILY     3.6     217      4.9M  10000     Paid  389.99  Everyone        Entertainment  June 22, 2018      1.5          4.2 and up      1
I Am Rich Premium               FINANCE    4.1     1867     4.7M  50000     Paid  399.99  Everyone        Finance        November 12, 2017  1.6          4.0 and up      1
I Am Rich Pro                   FAMILY     4.4     201      2.7M  5000      Paid  399.99  Everyone        Entertainment  May 30, 2017       1.54         1.6 and up      1
I am Rich                       FINANCE    4.3     180      3.8M  5000      Paid  399.99  Everyone        Finance        March 22, 2018     1.0       

b. Si metemos estas apps al modelo, la red neuronal creerá que vender a 400 dólares es un modelo de negocio viable. Sobrescriban el DataFrame conservando únicamente las apps que cuesten menos de 50 dólares (df = df[df['Price'] < 50]).

In [76]:
df = df[df['Price'] < 50]

c. Guarden su dataset final usando df.to_csv('playstore_limpio.csv', index=False).

In [77]:
df.to_csv('playstore_clean.csv', index=False)
print(f"Filas finales: {df.shape[0]}")

Filas finales: 9636
